src/NuGet.Clients/PackageManagement.VisualStudio/Telemetry/NuGetProjectTelemetryService.cs

```diff
@@ -0,0 +1,92 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Linq;
+using System.Threading;
+using EnvDTE;
+using Microsoft.VisualStudio.Shell;
+using NuGet.Common;
+using NuGet.PackageManagement.VisualStudio;
+using NuGet.ProjectManagement;
+using NuGet.ProjectManagement.Projects;
+
+namespace NuGet.PackageManagement.Telemetry
+{
+    public class NuGetProjectTelemetryService
+    {
+        public static NuGetProjectTelemetryService Instance = new NuGetProjectTelemetryService(
+            NuGetTelemetryService.Instance);
+
+        private readonly NuGetTelemetryService _nuGetTelemetryService;
+        private readonly string _nuGetVersion;
+
+        public NuGetProjectTelemetryService(NuGetTelemetryService telemetryService)
+        {
+            _nuGetVersion = ClientVersionUtility.GetNuGetAssemblyVersion();
+            _nuGetTelemetryService = telemetryService;
+        }
+
+        public void EmitNuGetProject(Project vsProject, NuGetProject nuGetProject, CancellationToken token)
```

Comment:
Agree. I totally missed that. 
Did you consider emitting events in truly unattended fashion within `ThreadHelper.JoinableTaskFactory.RunAsync`?